Sync external SD card pref title with that of in StorageUtils

ozbek opened the pull request:
This is how it looks now:

![screenshot_2015-03-01-21-47-14](https://cloud.githubusercontent.com/assets/452620/6430791/85e1df10-c05c-11e4-91e9-78f754aa3b17.png)

It feels a little confusing with external storage number (1) and the free space amount (59933) being shown side-by-side. Probably, free space amount needs to be shown inside parenthesis or otherwise made it visually distinguishable.

ahmedre commented:
i am confused about how your fix fixes this though - you are converting "External SD Card" to "External SD Card 1" in this patch - before your patch, i take it that we used to just display "External SD Card," right?

ozbek commented:
Yes, that's right. If you see the `StorageUtils`, that's how the preference is titled: _External SD Card 1_, _External SD Card 2_, etc.

Are there user devices with more than one SD card slots? Maybe it should be synced other way round and card numbers should be removed from `StorageUtils`?

In fact, I am not sure why `QuranSettingsFragment` is not using `StorageUtils` at this point and trying to discover mount points directly (probably should in order to avoid DRY).

ahmedre commented:
ah - i see - yes we're adding the number in `StorageUtils` indeed - so a few things:
1. i agree - any of that storage stuff should live in `StorageUtils`.
2. `StorageUtils` can be cleaned up - it has a bunch of 2.3 and below code that can now be removed.
3. ideally, formatting like this should live in strings - i.e `External SD Card %1$d (%2$d)` or something similar.

ozbek commented:
This is a quick clean-up and implementation of the points you have mentioned. Only tested on Note 4 and Nexus 5, no exhausted  tests are done.

Please let me know your opinion on the changes. I see at least other strings/locales should be fixed.

ahmedre commented on app/src/main/java/com/quran/labs/androidquran/util/StorageUtils.java:
```diff
@@ -256,11 +233,11 @@ public int getFreeSpace() {
-    public int getTotalSpace() {
-      return totalSpace;
-    }
+//    /**
```
remove it instead

ahmedre commented on app/src/main/java/com/quran/labs/androidquran/util/StorageUtils.java:
```diff
@@ -232,12 +202,19 @@ public Storage(String label, String mountPoint) {
+        bytesAvailable = stat.getAvailableBlocksLong() * stat.getBlockSizeLong();
+      } else {
+        //noinspection deprecation
+//        totalBytes = (long) stat.getBlockCount() * (long) stat.getBlockSize();
```
remove the stuff you commented out, we can re-add if we need it later

ahmedre commented on app/src/main/java/com/quran/labs/androidquran/util/StorageUtils.java:
```diff
@@ -222,7 +192,7 @@ private static boolean isExternalStorageEmulatedHoneycomb() {
     private String mountPoint;
     private int freeSpace;
-    private int totalSpace;
+//    private int totalSpace;
```
remove

ahmedre commented on app/src/main/java/com/quran/labs/androidquran/util/StorageUtils.java:
```diff
@@ -68,35 +78,11 @@
       // Follow Android SDCards naming conventions
-      if (Build.VERSION.SDK_INT < Build.VERSION_CODES.GINGERBREAD) {
-        list.add(new Storage(
-            context.getString(R.string.prefs_sdcard_auto),
```
this isn't used anymore, can we remove the string from the strings files? speaking of which, we need a utility to make this easier, it's a hassle right now :(

ahmedre commented on app/src/main/res/values/strings.xml:
```diff
@@ -168,7 +168,7 @@
     <string name="prefs_app_location_summary">Choose where to store Quran files.</string>
     <string name="prefs_sdcard_internal">Internal Storage</string>
-    <string name="prefs_sdcard_external">External SD Card</string>
+    <string name="prefs_sdcard_external">External SD Card %1$d</string>
```
either need to change it for other languages as well or remove the string from all non-English languages so people can add them again.

ahmedre commented on app/src/main/java/com/quran/labs/androidquran/util/StorageUtils.java:
```diff
@@ -33,34 +33,44 @@
-            if (!vold.contains(mount)){
-                toRemove.add(mount);
-            }
+        for (Iterator<String> iter = mounts.iterator(); iter.hasNext(); ) {
```
`foreach (String mount : mounts) {`

ahmedre commented:
just some minor suggestions, otherwise looks good to me. the note 4 has an external sdcard i guess, so that's the better use case. i don't have a 4.x+ phone that has an external sdcard so it's also difficult for me to test.

ozbek commented:
Removed the commented out code and fixed other warning Android Studio gave in `StorageUtils`. 

Added number reference to all `prefs_sdcard_external` translations except for Kurdish (values-ku). This one has mixture of Latin and Arabic text, so I could not figure out where to correctly put the wildcard. But I neither remove the translation (as you have suggested). The current state should be good without any runtime issues. The only thing is number reference will go void at `getString()` for Kurdish.

ahmedre commented:
jazakAllah khairan - i manually merged it.
